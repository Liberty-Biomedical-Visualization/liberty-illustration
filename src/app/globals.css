@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --accent-50: 224 247 252;
  --accent-100: 178 224 240;
  --accent-200: 128 200 232;
  --accent-300: 77 176 215;
  --accent-400: 26 153 206;
  --accent-500: 3 152 206;
  --accent-600: 2 128 165;
  --accent-700: 2 109 136;
  --accent-800: 2 90 107;
  --accent-900: 2 61 77;
  --accent-950: 1 44 58;
  --background: 255 255 255;
  --foreground: 23 23 23;
}

body {
  color: var(--foreground);
  background: var(--background);
  font-family: Arial, Helvetica, sans-serif;
}

@layer utilities {
  .text-balance {
    text-wrap: balance;
  }
}
