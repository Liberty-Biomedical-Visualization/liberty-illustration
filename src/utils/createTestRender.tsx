import { render } from "@testing-library/react";
import type { JSX, JSXElementConstructor } from "react";

/**
 * Returns a function that allows rendering Component with partial props while
 * falling back to defaultProps for testing purposes.
 */
export default function createTestRender<C extends JSXElementConstructor<P>, P>(
  Component: JSXElementConstructor<P> & C,
  defaultProps: JSX.LibraryManagedAttributes<C, P>,
) {
  return function (props: Partial<P> = {}) {
    const resolvedProps: JSX.IntrinsicAttributes &
      JSX.LibraryManagedAttributes<C, P> = { ...defaultProps, ...props };
    return render(<Component {...resolvedProps} />);
  };
}
