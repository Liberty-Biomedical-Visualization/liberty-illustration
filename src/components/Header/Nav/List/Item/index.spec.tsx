import { describe, expect, it, jest } from "@jest/globals";
import { screen } from "@testing-library/dom";
import userEvent from "@testing-library/user-event";

import createTestRender from "@/lib/createTestRender";

import NavListItem from "./index";

describe("NavListItem", () => {
  it("should display a list item ", () => {
    const children = "Portfolio";
    renderNavListItem({ children });
    const result = screen.getByRole("listitem");
    expect(result).toBeVisible();
  });

  it("should display a link named by the passed children", () => {
    const children = "Portfolio";
    renderNavListItem({ children });
    const result = screen.getByRole("link", { name: children });
    expect(result).toBeVisible();
  });

  it("should propagate the passed href to the link", () => {
    const href = "/home";
    renderNavListItem({ href });
    const result = screen.getByRole("link");
    expect(result).toHaveAttribute("href", href);
  });

  it("should call onClick when clicked", async () => {
    const onClick = jest.fn();
    renderNavListItem({ onClick });

    const user = userEvent.setup();
    const button = screen.getByRole("link");
    await user.click(button);

    expect(onClick).toHaveBeenCalled();
  });

  it("should pass className to the outermost element", () => {
    const className = "foo";
    const result = renderNavListItem({ className });
    expect(result.container.firstChild).toHaveClass(className);
  });

  it("should match the snapshot", () => {
    const result = renderNavListItem();
    expect(result.container).toMatchSnapshot();
  });
});

const renderNavListItem = createTestRender(NavListItem, {
  children: "Home",
  href: "/",
  onClick: () => {},
});
