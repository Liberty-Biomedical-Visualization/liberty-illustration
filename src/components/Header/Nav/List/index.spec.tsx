import { describe, expect, it, jest } from "@jest/globals";
import { screen } from "@testing-library/dom";
import userEvent from "@testing-library/user-event";

import createTestRender from "@/lib/createTestRender";

import NavList from "./index";

describe("NavList", () => {
  it("should display a list", () => {
    renderNavList();
    const result = screen.getByRole("list");
    expect(result).toBeVisible();
  });

  it("should display list items containing links from passed pages", () => {
    const pages = [
      { href: "/", name: "Home" },
      { href: "/portfolio", name: "Portfolio" },
    ];
    renderNavList({ pages });

    const listItems = screen.getAllByRole("listitem");

    const homeLink = screen.getByRole("link", { name: "Home" });
    const containsHomeLink = listItems.some((listItem) =>
      listItem.contains(homeLink),
    );
    expect(containsHomeLink).toBe(true);

    const portfolioLink = screen.getByRole("link", { name: "Portfolio" });
    const containsPortfolioLink = listItems.some((listItem) =>
      listItem.contains(portfolioLink),
    );
    expect(containsPortfolioLink).toBe(true);
  });

  it("should call onItemClick when link is clicked", async () => {
    const pages = [{ href: "/", name: "Home" }];
    const onItemClick = jest.fn();
    renderNavList({ onItemClick, pages });

    const user = userEvent.setup();
    const link = screen.getByRole("link");
    await user.click(link);

    expect(onItemClick).toHaveBeenCalled();
  });

  it("should pass className to the outermost element", () => {
    const className = "foo";
    const result = renderNavList({ className });
    expect(result.container.firstChild).toHaveClass(className);
  });

  it("should match the snapshot", () => {
    const result = renderNavList();
    expect(result.container).toMatchSnapshot();
  });
});

const renderNavList = createTestRender(NavList, {
  onItemClick: () => {},
  pages: [],
});
