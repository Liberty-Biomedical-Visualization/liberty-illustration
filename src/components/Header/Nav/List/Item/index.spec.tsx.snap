// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavListItem should match the snapshot 1`] = `
<div>
  <li
    class="mb-4 md:mb-0 text-2xl text-center"
  >
    <a
      class="active:text-accent-400 block hover:text-accent-600 md:inline md:p-0 md:text-lg md:transition md:uppercase p-4 text-accent-800"
      href="/"
    >
      Home
    </a>
  </li>
</div>
`;
